### PR TITLE
[Wallet] Add host of URI as title to WebViewScreen and fix deep links

### LINKED ISSUE #6416: Fixes for WebViewScreen
### Expected Behavior

- The host of the webpage being shown should appear on the header.
- If the webpage being served on the WebView fires a deeplink starting with `celo://`, Valora should handle it.

### Current Behavior

- The header is empty.
- Valora doesn't handle deep links fired within web pages.

### PR DESCRIPTION
**Pull request by gnardini**
### Description

- The `WebViewScreen` doesn't have a title. Adding the host of the URI as title in the header.
- When firing a deep link form the WebView, the link was not being fired. Added a handler to support that usecase. In the past I had tested against a custom HTML which did work, but when using a real webpage it did not. It works after this change.

### Other changes
- Added an origin param to the CELO rewards link to be able to track usage.

### Tested

Yes

### Related issues

- Fixes #6416

### CODE REVIEW

File: packages/mobile/src/app/WebViewScreen.tsx [hunk 1]
```diff
@@ -3,33 +3,54 @@ import { StackScreenProps } from '@react-navigation/stack'
 import * as React from 'react'
 import { StyleSheet, View } from 'react-native'
 import { WebView } from 'react-native-webview'
+import { ShouldStartLoadRequest } from 'react-native-webview/lib/WebViewTypes'
```
Comment (jeanregisser) [inline packages/mobile/src/app/WebViewScreen.tsx:6]: nit: No way to combine the 2 imports?

**Review by jeanregisser — APPROVED** [review]

File: packages/mobile/src/app/WebViewScreen.tsx (hunk 1 above)
Comment (gnardini) [reply packages/mobile/src/app/WebViewScreen.tsx:6]: I'm not sure how, VSCode added them this way, I didn't do it manually.

**Comment by Lss-Ankit** [comment]
Hi @gnardini just need a small help from you. can you please send push notification to below account to verify this task.
US Phone number: (757) 699-9899
Account No: 0x57c23f1f62760099ea67e72a64b5b572b0cb0eb7

**Comment by Lss-Ankit** [comment]
Hi @gnardini we have verified this on latest Android Internal Build V1.8.0(1004294330) & Test Flight Build V1.8.0(40) & observed following:
- User redirected to webview page inapp after tapping on start earnings button in Dynamic notification.
- Also Able to" see website-staging-dot-celo-testnet.wl.r.appspot.com " link on top.

**Query:**
- We have terms and condition page which is open in Safari or chrome is it expected ?

**Pull request merged**